### PR TITLE
Fix request switcher selection cycling

### PR DESCRIPTION
**Pull request by akivajgordon**
When cycling through the request switcher (opened on macOS via ⌘P), pressing the down arrow will correctly cycle through all of the matched requests as well as matched workspaces. However, pressing the up arrow would only cycle through the matched requests and would not allow me to select the matched workspaces. Reading the existing code, it's pretty clear why that's happening. With this change, you can cycle through all of the matched requests and matched workspaces when cycling in either direction.

changelog(Fixes): Fixed an issue with cycling through requests in the request switcher

### CODE REVIEW

File: packages/insomnia-app/app/ui/components/modals/utils.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const wrapToIndex = (index: number, maxCount: number) => {
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/modals/utils.ts:1]: any ideas for a better name, anyone?

File: packages/insomnia-app/app/ui/components/modals/__tests__/utils.test.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  it.each([
+    { index: 0, maxCount: 4, result: 0 },
+    { index: 1, maxCount: 4, result: 1 },
+    { index: 3, maxCount: 3, result: 0 },
+    { index: -1, maxCount: 3, result: 2 },
+    { index: -3, maxCount: 3, result: 0 },
+  ])('%p', ({ index, maxCount, result }) => {
+    expect(wrapToIndex(index, maxCount)).toBe(result);
+  });
+
```
Comment (dimitropoulos) [inline packages/insomnia-app/app/ui/components/modals/__tests__/utils.test.tsx:13]: there was another test for `-0`, but I couldn't decided if it's actually of real value.  thoughts?

```ts
  it('throws when max is negative', () => {
    const index = 1;
    const maxCount = -0;
    const execute = () => wrapToIndex(index, maxCount);
    expect(execute).toThrow();
  });
```

**Review by dimitropoulos — APPROVED** [review]
@gatzjames / @jackkav when you take a look, to test this:
1. open up insomnia on production
1. create a workspace with two requests
1. use a shortcut to open the quick switch modal (ctrl+p or command+p on mac)
1. observe that the up arrow only cycles through requests
1. observe that the down arrow cycles through both requests and workspaces
1. open up this PR
1. open the quick switcher again
1. observe that the up arrow and down arrow both wrap

**Review by filfreire — APPROVED** [review]
LGTM 👍 

Was able to reproduce the issue (e.g. This is how it looks like on `2022.1.0`):
Kapture 2022-03-09 at 09 46 40

This is how it looks like after the fix:

Kapture 2022-03-09 at 09 54 43

File: packages/insomnia-app/app/ui/components/modals/request-switcher-modal.tsx [hunk 3]
```diff
@@ -114,15 +115,8 @@ class RequestSwitcherModal extends PureComponent<Props, State> {
 
   _setActiveIndex(activeIndex: number) {
     const maxIndex = this.state.matchedRequests.length + this.state.matchedWorkspaces.length;
```
Comment (jackkav) [inline packages/insomnia-app/app/ui/components/modals/request-switcher-modal.tsx:117]: I see three issues, nothing is a show stopper but hopefully worth some consideration.
- exception case - exceptions for control flow is a smell and this is an early generalisation of a function that is only used in one place, a simpler approach would be to sanitise the input if it is of unknown origin. In this case it's a sum of array lengths and unlikely to be less than zero so no need to sanitise.
- utils - would be better kept beside its usage in the react component to aid readability.
- wrapToIndex - this could also be called wrapAround, but still the function could be just an assignment that caters to only the known case eg `const wrapAroundIndex = (activeIndex + maxIndex) % maxIndex`

In summary either of the following would be good for me

1 . keep the test and point it at a one line function in the react component. 
2. remove the test and util and just use something like the above one-liner with a comment like `// keep activeIndex within maxIndex bounds`

Comment (akivajgordon) [reply packages/insomnia-app/app/ui/components/modals/request-switcher-modal.tsx:117]: @jackkav – thanks for the feedback. Regarding your second point, I'll refer you to [my original commit](https://github.com/Kong/insomnia/pull/4519/commits/a03cae9c808c08cb195f245b0a6232fb56bf3935), which was pretty simple.

It was during the livestream that the utils and tests were added. @dimitropoulos – can I tag you in here to duke out the merits of the tests with @jackkav? At the risk of sounding selfish or apathetic, I don't have as much stake in the matter as do the maintainers of the project, so it feels like a discussion that's more relevant to you.

Comment (dimitropoulos) [reply packages/insomnia-app/app/ui/components/modals/request-switcher-modal.tsx:117]: @akivajgordon we're going to be streaming again in an hour or so, I'll address this feedback first and maybe we can get it merged today!
